plugins/discovery-ec2/src/main/java/org/opensearch/discovery/ec2/Ec2DiscoveryPlugin.java

```diff
@@ -161,6 +159,9 @@ static Settings getAvailabilityZoneNodeAttributes(Settings settings, String azMe
         final URLConnection urlConnection;
         try {
             url = new URL(azMetadataUrl);
+            // Obtain the current EC2 instance availability zone from IMDS.
+            // Same as curl http://169.254.169.254/latest/meta-data/placement/availability-zone/.
+            // TODO: use EC2MetadataUtils::getAvailabilityZone directly which is a similar implementation
```

Comment:
Does this TODO need to be addressed?

Reply:
Changed this to "TODO: use EC2MetadataUtils::getAvailabilityZone that was added in AWS SDK v2 instead of rolling our own". AWS SDKv2 has a similar implementation to the code we have here, but also has other "features" like caching. I didn't feel sufficiently confident to delete our implementation in favor of the one available in the SDK.